gnomad/assessment/sanity_checks.py

```diff
@@ -103,90 +117,850 @@ def make_filters_sanity_check_expr(
     return filters_dict
 
 
-def sample_sum_check(
-    ht: hl.Table,
-    prefix: str,
+def make_group_sum_expr_dict(
+    t: Union[hl.MatrixTable, hl.Table],
+    subset: str,
     label_groups: Dict[str, List[str]],
-    verbose: bool,
-    subpop: bool = None,
     sort_order: List[str] = SORT_ORDER,
-) -> None:
+    delimiter: str = "-",
+    metric_first_label: bool = True,
+    metrics: List[str] = ["AC", "AN", "nhomalt"],
+) -> Dict[str, Union[hl.expr.Int32Expression, hl.expr.StructExpression]]:
     """
-    Compute the sum of annotations for a specified group of annotations, and compare to the annotated version.
+    Compute the sum of call stats annotations for a specified group of annotations, compare to the annotated version, and display the result in stdout.
 
-    Results from checking the sum of the specified annotations are printed in the terminal.
+    For example, if subset1 consists of pop1, pop2, and pop3, check that t.info.AC-subset1 == sum(t.info.AC-subset1-pop1, t.info.AC-subset1-pop2, t.info.AC-subset1-pop3).
 
-    :param ht: Table containing annotations to be summed.
-    :param prefix: String indicating sample subset.
-    :param label_groups: Dictionary containing an entry for each label group, where key is the name of the grouping,
-        e.g. "sex" or "pop", and value is a list of all possible values for that grouping (e.g. ["male", "female"] or ["afr", "nfe", "amr"]).
-    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False,
-        show only top values of annotations that fail checks.
-    :param subpop: Subpop abbreviation, supplied only if subpopulations are included in the annotation groups being checked.
+    :param t: Input MatrixTable or Table containing call stats annotations to be summed.
+    :param subset: String indicating sample subset.
+    :param label_groups: Dictionary containing an entry for each label group, where key is the name of the grouping, e.g. "sex" or "pop", and value is a list of all possible values for that grouping (e.g. ["XY", "XX"] or ["afr", "nfe", "amr"]).
     :param sort_order: List containing order to sort label group combinations. Default is SORT_ORDER.
-    :return: None
+    :param delimiter: String to use as delimiter when making group label combinations. Default is "-".
+    :param metric_first_label: If True, call statistic precedes label group, e.g. AC-subset-male. If False, label group precedes metric, subset-AC-male. Default is True.
```

Comment:
Yeah I think the variable name is actually misleading since `label` is used in the function but has nothing to do with the variable `metric_first_label`. I propose this is changed to `metric_first_field` to better represent what is going on and I will update the docstring again to better explain it. Thoughts?